### PR TITLE
chore: Refactor templates into subfolders

### PR DESCRIPTION
**Pull request by mkilchhofer**
I already asked in the [Slack channel](https://kubernetes.slack.com/archives/C01JGDP8MB8/p1683745767910319) about this:

> image

As KEDA evolved the chart can be grouped into the 3 roles of KEDA for better understanding how things are put together.

Although there are a bit other role names in the [documentation](https://keda.sh/docs/2.10/concepts/#architecture) (Agent / Metrics / Admission Webhooks), I reused what's inside the [core repo](https://github.com/kedacore/keda/tree/main/config):
- manager
- metrics-server
- webhooks

### Checklist

- [x] I have verified that my change is according to the [deprecations & breaking changes policy](https://github.com/kedacore/governance/blob/main/DEPRECATIONS.md)
- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/charts/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))

### CODE REVIEW

File: keda/Chart.yaml [hunk 1]
```diff
@@ -8,7 +8,7 @@ kubeVersion: ">=v1.24.0-0"
 
 # This is the chart version. This version number should be incremented each time you make changes
 # to the chart and its templates, including the app version.
-version: 2.10.2
+version: 2.10.3
```
Comment (tomkerkhove) [inline keda/Chart.yaml:11]: Can you please revert this?

**Review by tomkerkhove — APPROVED** [review]
Interesting, I did not know this was supported - Thanks

**Review by JorTurFer — APPROVED** [review]

File: keda/Chart.yaml (hunk 1 above)
Comment (mkilchhofer) [reply keda/Chart.yaml:11]: sure :)

Comment (mkilchhofer) [reply keda/Chart.yaml]: done 👍

**Pull request merged**